### PR TITLE
community[patch]: VoyageAI embedding with input_type parameter

### PR DESCRIPTION
**Pull request by nicolas-geysse**
This pull request adds functionality to make the `inputType` field optional in the `VoyageEmbeddings` class. The changes include:

- Updating the `VoyageEmbeddingsParams` interface to include `inputType` as optional.
- Updating the `VoyageEmbeddings` class to handle `inputType` as an optional parameter.
- Adding tests to verify the correct handling of `inputType`.
- Updating the documentation to include examples of usage with and without `inputType`.

These changes allow users to specify `inputType` as `query`, `document`, or leave it undefined to use the default value `None`.

Fixes # (issue)

### CODE REVIEW

File: docs/core_docs/docs/integrations/text_embedding/voyageai.mdx [hunk 1]
```diff
@@ -1,11 +1,19 @@
...
+- `document`: Use this for documents or content that you want to be retrievable. Voyage AI will prepend a prompt to optimize the embeddings for document use cases.
+- `None` (default): The input text will be directly encoded without any additional prompt.
+
+
 ```typescript
 import { VoyageEmbeddings } from "langchain/embeddings/voyage";
 
 const embeddings = new VoyageEmbeddings({
   apiKey: "YOUR-API-KEY", // In Node.js defaults to process.env.VOYAGEAI_API_KEY
+  inputType: "None", // Optional: specify input type as 'query', 'document', or omit for None
```
Comment (jacoblee93) [inline docs/core_docs/docs/integrations/text_embedding/voyageai.mdx:17]: `None` is weird here?

File: libs/langchain-community/src/embeddings/voyage.ts [hunk 2]
```diff
@@ -41,7 +54,7 @@ export class VoyageEmbeddings
   extends Embeddings
   implements VoyageEmbeddingsParams
 {
-  modelName = "voyage-01";
+  modelName = "voyage-2";
```
Comment (jacoblee93) [inline libs/langchain-community/src/embeddings/voyage.ts:57]: This is technically breaking - people have complained about this before. Can you revert?

File: docs/core_docs/docs/integrations/text_embedding/voyageai.mdx [hunk 3]
```diff
@@ -1,11 +1,19 @@
-# Voyage AI
+## Voyage AI Integration
```
Comment (jacoblee93) [inline docs/core_docs/docs/integrations/text_embedding/voyageai.mdx:1]: Revert title change

**Comment by jacoblee93** [comment]
Thanks! Will close your other one

File: docs/core_docs/docs/integrations/text_embedding/voyageai.mdx (hunk 1 above)
Comment (nicolas-geysse) [reply docs/core_docs/docs/integrations/text_embedding/voyageai.mdx:17]: From what I read here :
https://docs.voyageai.com/docs/embeddings
"When the input_type is set to None, and the input text will be directly encoded by our embedding model. Alternatively, when the inputs are documents or queries, the users can specify input_type to be query or document, respectively. "
So sounds good to me ?

File: libs/langchain-community/src/embeddings/voyage.ts (hunk 2 above)
Comment (nicolas-geysse) [reply libs/langchain-community/src/embeddings/voyage.ts:57]: How can I accept your change ? ^^

**Review by nicolas-geysse — COMMENTED** [review]
ok !

File: libs/langchain-community/src/embeddings/voyage.ts [hunk 4]
```diff
@@ -32,6 +37,14 @@ export interface CreateVoyageEmbeddingRequest {
    * @memberof CreateVoyageEmbeddingRequest
    */
   input: string | string[];
+  
+  /**
+   * Input type for the embeddings request.
+   * @type {string}
+   * @memberof CreateVoyageEmbeddingRequest
+   */
+
```
Comment (jacoblee93) [inline libs/langchain-community/src/embeddings/voyage.ts:46]: nit: Remove additional space, and `@type` and `@memberof` are unnecessary

**Review by nicolas-geysse — COMMENTED** [review]
done

**Comment by jacoblee93** [comment]
@nicolas-geysse I think they mean None in the Python sense:

https://www.w3schools.com/python/ref_keyword_none.asp

In JS, this is the same as undefined

**Review by nicolas-geysse — COMMENTED** [review]
ok

**Review by nicolas-geysse — COMMENTED** [review]
done

**Comment by jacoblee93** [comment]
I can land from here thank you!

**Comment by jacoblee93** [comment]
Thanks again!

**Comment by nicolas-geysse** [comment]
> Thanks again!

Thanks a lot to you, that was really epic for me ^^

**Pull request merged**